pingcap/tidb
# Makefile: fix build datetime

### What problem does this PR solve?

Issue Number: Fixes #19077

Problem Summary:

The build timestamp was a 12 hour clock (format "I"), without an AM / PM specified. I changed it to 24 hour clock (format "H"). See: https://www.thegeekstuff.com/2013/05/date-command-examples/ for examples.

It's a minor annoyance being inaccurate, I don't think it needs cherry-picking if nobody has spotted it until now!

### What is changed and how it works?

What's Changed:

I changed it to 24 hr clock, because I thought this would be the most backwards compatible for anyone parsing this date. 

### Related changes

None

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

```
nullnotnil@ubuntu:~/go/src/github.com/nullnotnil/tidb$ date -u
Fri 07 Aug 2020 07:32:17 PM UTC
nullnotnil@ubuntu:~/go/src/github.com/nullnotnil/tidb$ make
CGO_ENABLED=1 GO111MODULE=on go build  -tags codes  -ldflags '-X "github.com/pingcap/parser/mysql.TiDBReleaseVersion=v4.0.0-beta.2-914-g503011a66" -X "github.com/pingcap/tidb/util/versioninfo.TiDBBuildTS=2020-08-07 19:32:21" -X "github.com/pingcap/tidb/util/versioninfo.TiDBGitHash=503011a66c54cc33247ec95fd99200ca89c8a2c3" -X "github.com/pingcap/tidb/util/versioninfo.TiDBGitBranch=fix-build-time" -X "github.com/pingcap/tidb/util/versioninfo.TiDBEdition=Community" ' -o bin/tidb-server tidb-server/main.go
Build TiDB Server successfully!
nullnotnil@ubuntu:~/go/src/github.com/nullnotnil/tidb$ ./bin/tidb-server  -V
Release Version: v4.0.0-beta.2-914-g503011a66
Edition: Community
Git Commit Hash: 503011a66c54cc33247ec95fd99200ca89c8a2c3
Git Branch: fix-build-time
UTC Build Time: 2020-08-07 19:32:21
GoVersion: go1.14.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

Side effects

- None

### Release note 

- None
